### ISSUE TITLE
修复 Dapper.Contrib.Extensions 中 Insert 方法，此方法返回值为long类型，但内部各类型数据库获取自增ID，都是转换为int32，当ID超过超过Int32（2147483647）时，出现bug。

### ISSUE THREAD

**Issue by ypeuee** [DapperLib/Dapper.Contrib#4]
修复 Dapper.Contrib.Extensions 中 Insert 方法，此方法返回值为long类型，但内部各类型数据库获取自增ID，都是转换为int32，当ID超过超过Int32（2147483647）时，出现bug。

主要涉及到的内容为接口 ISqlAdapter 中的方法：

`
/// <summary>
/// The interface for all Dapper.Contrib database operations
/// Implementing this is each provider's model.
/// </summary>
public partial interface ISqlAdapter
{
    /// <summary>
    /// Inserts <paramref name="entityToInsert"/> into the database, returning the Id of the row created.
    /// </summary>
    /// <param name="connection">The connection to use.</param>
    /// <param name="transaction">The transaction to use.</param>
    /// <param name="commandTimeout">The command timeout to use.</param>
    /// <param name="tableName">The table to insert into.</param>
    /// <param name="columnList">The columns to set with this insert.</param>
    /// <param name="parameterList">The parameters to set for this insert.</param>
    /// <param name="keyProperties">The key columns in this table.</param>
    /// <param name="entityToInsert">The entity to insert.</param>
    /// <returns>The Id of the row created.</returns>
    int Insert(IDbConnection connection, IDbTransaction transaction, int? commandTimeout, string tableName, string columnList, string parameterList, IEnumerable<PropertyInfo> keyProperties, object entityToInsert);

....
`

，此方法返回值改为long可修复此问题。

**Comment by benny856694** [comment]
exactly, we want this update, especially when working with sqlite, it's id can be as big as int64.

**Comment by wswind** [comment]
this is an old issue for dapper.contrib, it's a breaking change so dapper.contrib won't merge the pr:
<https://github.com/DapperLib/Dapper/pull/1386>

the work around is to insert a list with one item instead of  single entity for this situation, the return value would be row count instead of primary key.

check the code of dapper contrib :
https://github.com/DapperLib/Dapper.Contrib/blob/cf24f6bdc577b1e071c3764ddfb2cf3382531405/src/Dapper.Contrib/SqlMapperExtensions.cs#L380-L385

**Comment by mikesigs** [comment]
@wswind How we get the value of the new PK then? It's not _just_ that the method returns the PK value, it also assigns it to the Id property on the provided object.

**Comment by wswind** [comment]
@mikesigs  use dapper only, write the sql yourself.  For sqlserver it's something like this:  `insert into ...
output inserted.ID values ... ` , then you can assign the id yourself.  This is just a workaround.
